### DISCUSSION TITLE
NextJS and WebPack are Painfully Slow

### DISCUSSION THREAD

**Discussion by simmbiote** [vercel/next.js#15220] [Help]
Hi

I'm finding it quite painful to get things done while waiting for NextJS/Webpack to start running (ie. `npm run dev`) and then to reload, and then to build at the end. It feels like in 30 minutes I probably sometimes get 5 minutes worth of new code in. Waiting 5+minutes for things to happen. How can I fix this?

I've used Speed Measure Webpack Plugin (npm:  speed-measure-webpack-plugin) to determine what is slowing things down; this is what it found:

![image](https://user-images.githubusercontent.com/11704483/87667427-9c3e2280-c76a-11ea-8ae1-5de06023db62.png)

 SMP  ⏱
General output time took 19.35 secs

 SMP  ⏱  Plugins
**AutoDLLPlugin took 52.58 secs**
......

 SMP  ⏱  Loaders
**modules with no loaders took 14.57 secs**
.....
This is just the beginning - loading in the browser takes a further eternity.

So far:
I've upgraded from NextJS 9.3.6 to 9.4.4 - no differences
I've tried upgrading WebPack to v5 beta, but NextJS doesn't support this yet, so it's not an option.

Any ideas? 🙏 

**NPM packages:**

 "dependencies": {
    "@babel/core": "^7.10.5",
    "@babel/polyfill": "^7.10.4",
    "@babel/runtime": "^7.10.5",
    "@material-ui/core": "4.9.13",
    "@material-ui/icons": "4.9.1",
    "@material-ui/lab": "^4.0.0-alpha.56",
    "@zeit/next-sass": "1.0.1",
    "amazon-cognito-auth-js": "^1.3.3",
    "classnames": "2.2.6",
    "core-js": "^2.6.5",
    "fast-xml-parser": "^3.17.4",
    "he": "^1.2.0",
    "isomorphic-fetch": "^2.2.1",
    "jsonwebtoken": "^8.5.1",
    "moment": "2.25.3",
    "next": "9.4.4",
    "next-compose-plugins": "2.2.0",
    "next-images": "1.3.1",
    "nextjs-redirect": "2.0.2",
    "node-sass": "4.14.1",
    "nouislider": "14.4.0",
    "path": "0.12.7",
    "promise-polyfill": "^8.1.3",
    "prop-types": "15.7.2",
    "react": "16.13.1",
    "react-barcode": "^1.4.0",
    "react-datetime": "2.16.3",
    "react-dom": "16.13.1",
    "react-idle": "^3.0.0",
    "react-slick": "0.25.2",
    "react-swipeable-views": "0.13.9",
    "regenerator-runtime": "^0.13.5",
    "universal-cookie": "^4.0.3",
    "webpack": "^4.43.0"
  },
  "devDependencies": {
    "cz-customizable": "^6.2.1",
    "speed-measure-webpack-plugin": "^1.3.3",
    "standard-version": "^8.0.2"
  }

**.babelrc (IE support is required)**

{
  "presets": [
    ["next/babel"],
    [
      "@babel/preset-env",
      {
        "useBuiltIns": "entry",
        "corejs": "2",
        "targets": {
          "browsers": ["> 1%", "last 1 versions", "ie 11"]
        }
      }
    ]
  ],
  "plugins": ["@babel/plugin-transform-runtime"]
}

**Comment by jamesmosier** [comment]
Do you have a repository that you could share that demonstrates this?

Also Next includes all of the Babel plugins/env options you have by default. Next fully supports IE11 so including corejs is not needed as well as [babel/plugin-transform-runtime](https://github.com/vercel/next.js/blob/b124ed2e143993c6b2198c3eaae6bbaadbaea8c2/packages/next/build/babel/preset.ts#L158-L158) is used in Next by default. So your entire `.babelrc` could be removed.

**Reply by simmbiote** [reply]
Thanks James, can't share the repo unfortunately, but I can say that it's based on this theme/boilerplate: https://github.com/creativetimofficial/nextjs-material-kit. 
Anything particular you want to look at?

After removing babel/core and .babelrc file, here's the new Speed info. It's slower than before?

![image](https://user-images.githubusercontent.com/11704483/87779035-74af8e80-c82c-11ea-95cc-08bbcbc5732e.png)

After stopping and starting again, results seem much better.
![image](https://user-images.githubusercontent.com/11704483/87779401-03bca680-c82d-11ea-86a1-21515805c941.png)

At this point waiting for something to happen in the browser - nothing shows in the Network tab. It just spins...
![image](https://user-images.githubusercontent.com/11704483/87779696-804f8500-c82d-11ea-92ef-2fa944c8ffec.png)

Eventually after about 15 mins it's ready
![image](https://user-images.githubusercontent.com/11704483/87781180-809d4f80-c830-11ea-8660-b24613f845df.png)


Here is next.config.js in case it's useful:

```
const withPlugins = require("next-compose-plugins");
const withImages = require("next-images");
const withSass = require("@zeit/next-sass");
const webpack = require("webpack");
const path = require("path");


const SpeedMeasurePlugin = require("speed-measure-webpack-plugin");
const smp = new SpeedMeasurePlugin();

module.exports = withPlugins([[withSass], [withImages]], {
  webpack(config, options) {
    config.resolve.modules.push(path.resolve("./"));

    // for use SpeedMeasurePlugin
    return smp.wrap(config);
    // return config;
  },
});
```

**Reply by jamesmosier** [reply]
Sorry, unfortunately without seeing the code it's hard for me to know where to start. My go-to answer would be to delete your `.next` and `node_modules` folders and try running `yarn dev` (or `npm run dev`) again.

Short of that I've never seen this issue so it may be system related unless you can have someone else recreate it on their machine.

**Comment by IRediTOTO** [comment]
I think problem come from your code not nextjs. Lets try to check more. I got this before, take some hours to find my stupid :D

**Reply by simmbiote** [reply]
Hi
The code is largely based on this boilerplate:
https://github.com/creativetimofficial/nextjs-material-kit
The main feature is Material UI - I wonder if that's the issue?
I noticed that my CPU usage reaches 100% when I have these problems - Chrome is always shown as the culprit. I'm on 8GB ram (Windozzze).

**Comment by arthurva** [comment]
Hi @simmbiote ,
Have you found a solution?
